main.c

```diff
@@ -182,36 +190,66 @@ static const struct wl_pointer_listener pointer_listener = {
 	.axis = noop,
 };
 
+static void keyboard_handle_keymap(void *data, struct wl_keyboard *wl_keyboard,
+		uint32_t format, int32_t fd, uint32_t size) {
+	struct slurp_seat *seat = data;
+	switch (format) {
+	case WL_KEYBOARD_KEYMAP_FORMAT_NO_KEYMAP:
+		seat->xkb_keymap = xkb_keymap_new_from_names(seat->state->xkb_context, NULL, XKB_KEYMAP_COMPILE_NO_FLAGS);
+		break;
+	case WL_KEYBOARD_KEYMAP_FORMAT_XKB_V1:;
+		void *buffer;
+		if ((buffer = mmap(NULL, size - 1, PROT_READ, MAP_SHARED, fd, 0)) == MAP_FAILED) {
```

Comment:
No need for this `- 1` which could underflow.

Reply:
https://github.com/xkbcommon/libxkbcommon/blob/master/test/interactive-wayland.c#L341

it uses -1 but not for mmap (which is an inconsistency from the example). I guess the buffer should be a valid string so it must have a size of at least 1 (for the '\0').

Here it also uses the length without the '\0' https://github.com/xkbcommon/libxkbcommon/blob/d92a248c48227d09f6fdcfafaf339a5ff586e042/src/keymap.c#L164